constructor/header.sh

```diff
@@ -21,6 +21,32 @@ if ! echo "$0" | grep '\.sh$' > /dev/null; then
     return 1
 fi
 
+#if osx and min_osx_version
+min_osx_version="__MIN_OSX_VERSION__"
+system_osx_version=$(SYSTEM_VERSION_COMPAT=0 sw_vers -productVersion)
+# shellcheck disable=SC2183 disable=SC2046
+int_min_osx_version="$(printf "%02d%02d%02d" $(echo "$min_osx_version" | sed 's/\./ /g'))"
+# shellcheck disable=SC2183 disable=SC2046
+int_system_osx_version="$(printf "%02d%02d%02d" $(echo "$system_osx_version" | sed 's/\./ /g'))"
+if [  "$int_system_osx_version" -lt "$int_min_osx_version" ]; then
+    echo "Installer requires macOS >=${min_osx_version}, but system has ${system_osx_version}."
+    exit 1
+fi
+#endif
+#if linux and min_glibc_version
+min_glibc_version="__MIN_GLIBC_VERSION__"
+# ldd reports glibc in the last field of the first line
+system_glibc_version=$(ldd --version | awk 'NR==1{print $NF}')
```

Comment:
`ldd` on an alpine docker image with glibc installed will have a different output. `ldd` here would give musl version and not glibc. `/lib64/ld-linux-x86-64.so.2 --version` might be better.

Reply:
You can use https://github.com/Docker-Hub-frolvlad/docker-alpine-glibc if you want to test.

Reply:
Thanks Isuru! Added a workaround for musl systems. Invoking ld.so directly in CentOS does not work so I'll leave it with those two branches for now.